hail/python/hailtop/batch/genetics/regenie/regenie.py

```diff
@@ -0,0 +1,314 @@
+from hailtop.batch import Batch, LocalBackend
+from hailtop.batch.resource import Resource
+from collections import namedtuple
+import sys
+import shlex
+from argparse import Namespace, ArgumentParser
+from typing import Set
+from os.path import splitext, basename, exists
+from google.cloud import storage
+from google.cloud.storage.blob import Blob
+
+
+BatchArgs = namedtuple("BatchArgs", ['cores', 'memory', 'storage'])
+input_file_args = ["bgen", "bed", "pgen", "sample", "keep", "extract", "exclude", "remove",
+                   "phenoFile", "covarFile"]
+
+from_underscore = {
+    "force_impute": "force-impute",
+    "ignore_pred": "ignore-pred",
+    "lowmem_prefix": "lowmem-prefix"
+}
+
+
+def _is_local(spath: str):
+    if(spath.startswith("gs://")):
+        return False
+    return True
+
+
+def _read(spath: str):
+    if _is_local(spath):
+        with open(spath, "r") as f:
+            return f.read()
+
+    client = storage.Client()
+    blob = Blob.from_string(spath, client)
+    return blob.download_as_string()
+
+
+def _read_first_line(spath: str):
```

Comment:
This is a waste of code to maintain. GoogleCloudStorageFS is available to do exactly what I'm replicating here.